debugger/loadprinter.ml

```diff
@@ -124,10 +124,20 @@ let install_printer ppf lid =
       raise(Error(Unavailable_module(s, lid))) in
   let print_function =
     if is_old_style then
-      (fun _formatter repr -> Obj.obj v (Obj.obj repr))
+      (fun _ppf remote_val ->
+         try
+           Obj.obj v (Debugcom.Remote_value.obj remote_val)
+         with
+           Debugcom.Marshalling_error ->
+             fprintf ppf "<cannot fetch remote object>")
     else
```

Comment:
I'm not fond of the redundancy introduced by the two branches of this if-then-else. Could you move the test within the `try..with` branch to share the exception handler?